src/vs/workbench/contrib/preferences/browser/settingsTree.ts

```diff
@@ -1322,7 +1322,8 @@ abstract class AbstractSettingObjectRenderer extends AbstractSettingRenderer imp
 			}
 
 			if (template.onChange) {
-				template.onChange(newValue);
+				const valueToReport = Object.keys(newValue).length ? newValue : undefined;
```

Comment:
Should this be comparing against the default? I would think that you want to do the reset if the new value is the same as the default. Otherwise if the default has some boxes checked, and I uncheck them, it will do a reset right?

Reply:
For the bool object, values that match the default are kicked off from `newValue`, so we'd be left with an empty object for `newValue`. But, this line causes a problem when we're dealing with non-boolean objects.